### PR TITLE
Add serialisationbenches

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: sn_client/Cargo.toml [hunk 1]
```diff
@@ -45,7 +49,7 @@ bls = { package = "blsttc", version = "7.0.0" }
 bls_dkg = "~0.10.5"
 bytes = { version = "1.0.1", features = ["serde"] }
 clap = { version = "3.0.0", features = ["derive"], optional = true }
-crdts = { version = "7.2", default-features = false, features = ["merkle"] }
+crdts = { version = "7.1", default-features = false, features = ["merkle"] }
```
Comment (maqi) [inline sn_client/Cargo.toml:52]: just to confirm, so those three crates are for `downgrade` ?

File: sn_client/benches/serialize.rs [hunk 2]
```diff
@@ -0,0 +1,136 @@
+// Copyright 2021 MaidSafe.net limited.
```
Comment (maqi) [inline sn_client/benches/serialize.rs:1]: 2022

File: .github/workflows/benchmark_prs.yml [hunk 3]
```diff
@@ -46,6 +46,17 @@ jobs:
       - name: ubuntu install ripgrep
         run: sudo apt-get -y install ripgrep
 
+      #############################
+      ### Node Benchmark itself ###
+      #### Do not need network ####
+      #############################
```
Comment (maqi) [inline .github/workflows/benchmark_prs.yml:52]: just to confirm, this `highlight` is just to alert, right?
nothing related to the following new bench test in sn_client

File: sn_client/benches/serialize.rs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+        let (_address, batch) = match client.create_register(name, tag, policy).await {
+            Ok(x) => x,
+            Err(error) => panic!("error creating register {error:?}"),
+        };
+
+        let client_pk = client.public_key();
+
+        // batch[0].clone()
+        // let payload = ServiceMsg::Cmd(batch[0].clone());
```
Comment (maqi) [inline sn_client/benches/serialize.rs:72]: remove?

File: sn_client/benches/serialize.rs [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    // so we just use this to get our throughput measurement
+    let msg_bytes = match the_wire_msg.serialize() {
+        Ok(bytes) => bytes,
+        Err(_error) => panic!("Could not serialize the msg bytes"),
+    };
+
+    group.throughput(Throughput::Bytes(
+        (std::mem::size_of_val(&msg_bytes) * dsts.len()) as u64,
+    ));
+    // upload and read
```
Comment (maqi) [inline sn_client/benches/serialize.rs:114]: just serialise X copies of msgs with different dsts ?

File: sn_client/Cargo.toml (hunk 1 above)
Comment (joshuef) [reply sn_client/Cargo.toml:52]: ah, no, that should not be. That's probs crept in by accident over some file copying

File: sn_client/benches/serialize.rs (hunk 5 above)
Comment (joshuef) [reply sn_client/benches/serialize.rs:114]: Yeh, at the moment we reserialize the entire header for just a dst change. This is one place where we should see improvements with upcoming PRs

**Review by maqi — APPROVED** [review]

**Pull request merged**